internal/cloudapi/server.go

```diff
@@ -50,94 +63,81 @@ func NewServer(workers *worker.Server, rpmMetadata rpmmd.RPMMD, distros *distror
-		const identityHeaderKey contextKey = iota
-		type identityHeader struct {
-			Identity struct {
-				AccountNumber string `json:"account_number"`
-			} `json:"identity"`
-		}
-
-		idHeaderB64 := r.Header["X-Rh-Identity"]
+func (server *Server) VerifyIdentityHeader(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		idHeaderB64 := c.Request().Header.Get("X-Rh-Identity") //r.Header["X-Rh-Identity"]
 		if len(idHeaderB64) != 1 {
-			http.Error(w, "Auth header is not present", http.StatusNotFound)
-			return
+			return echo.NewHTTPError(http.StatusNotFound, "Auth header is not present")
 		}
 
-		b64Result, err := base64.StdEncoding.DecodeString(idHeaderB64[0])
+		b64Result, err := base64.StdEncoding.DecodeString(idHeaderB64)
 		if err != nil {
-			http.Error(w, "Auth header has incorrect format", http.StatusNotFound)
-			return
+			return echo.NewHTTPError(http.StatusNotFound, "Auth header has incorrect format")
 		}
 
 		var idHeader identityHeader
 		err = json.Unmarshal([]byte(strings.TrimSuffix(fmt.Sprintf("%s", b64Result), "\n")), &idHeader)
 		if err != nil {
-			http.Error(w, "Auth header has incorrect format", http.StatusNotFound)
-			return
+			return echo.NewHTTPError(http.StatusNotFound, "Auth header has incorrect format")
 		}
 
 		for _, i := range server.identityFilter {
 			if idHeader.Identity.AccountNumber == i {
-				next.ServeHTTP(w, r.WithContext(context.WithValue(r.Context(), identityHeaderKey, idHeader)))
-				return
+				c.Set("IdentityHeader", idHeader)
+				c.Set("IdentityHeaderKey", identityHeaderKey)
+				return next(c)
 			}
 		}
-		http.Error(w, "Account not allowed", http.StatusNotFound)
-	})
+		return echo.NewHTTPError(http.StatusNotFound, "Account not allowed")
+	}
 }
 
-func (s *Server) IncRequests(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func (s *Server) IncRequests(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
 		prometheus.TotalRequests.Inc()
-		if strings.HasSuffix(r.URL.Path, "/compose") {
+		if strings.HasSuffix(c.Path(), "/compose") {
 			prometheus.ComposeRequests.Inc()
 		}
-		next.ServeHTTP(w, r)
-	})
+		return next(c)
+	}
 }
 
 // Compose handles a new /compose POST request
-func (server *Server) Compose(w http.ResponseWriter, r *http.Request) {
-	contentType := r.Header["Content-Type"]
+func (h *apiHandlers) Compose(ctx echo.Context) error {
+	contentType := ctx.Request().Header["Content-Type"]
 	if len(contentType) != 1 || contentType[0] != "application/json" {
-		http.Error(w, "Only 'application/json' content type is supported", http.StatusUnsupportedMediaType)
-		return
+		return echo.NewHTTPError(http.StatusUnsupportedMediaType, "Only 'application/json' content type is supported")
 	}
 
 	var request ComposeRequest
-	err := json.NewDecoder(r.Body).Decode(&request)
+	err := json.NewDecoder(ctx.Request().Body).Decode(&request)
```

Comment:
We should really do this in a binder, see [kojiapi for reference](https://github.com/osbuild/osbuild-composer/blob/main/internal/kojiapi/server.go#L490). But could also do this in a follow-up.

Reply:
Got it, thank you! I think I'll change this in a follow-up, at the moment I really want to get this PR in